Sources/Rendering/OpenGL/CustomPostProcessingPass/index.js

```diff
@@ -0,0 +1,317 @@
+          ''
+        );
+      const program = model.convolutionShader;
+
+      // prepare the vertex and triangle data for the image plane to render to
+      model.copyVAO = vtkVertexArrayObject.newInstance();
+      model.copyVAO.setOpenGLRenderWindow(viewNode);
+
+      model.tris.getCABO().bind();
+      if (
+        !model.copyVAO.addAttributeArray(
+          program,
+          model.tris.getCABO(),
+          'vertexDC',
+          model.tris.getCABO().getVertexOffset(),
+          model.tris.getCABO().getStride(),
+          gl.FLOAT,
+          3,
+          gl.FALSE
+        )
+      ) {
+        vtkErrorMacro('Error setting vertexDC in copy shader VAO.');
+      }
+      if (
+        !model.copyVAO.addAttributeArray(
+          program,
+          model.tris.getCABO(),
+          'tcoordTC',
+          model.tris.getCABO().getTCoordOffset(),
+          model.tris.getCABO().getStride(),
+          gl.FLOAT,
+          2,
+          gl.FALSE
+        )
+      ) {
+        vtkErrorMacro('Error setting vertexDC in copy shader VAO.');
+      }
+    } else {
+      viewNode.getShaderCache().readyShaderProgram(model.convolutionShader);
+    }
+
+    gl.viewport(0, 0, size[0], size[1]);
+    gl.scissor(0, 0, size[0], size[1]);
+
+    // activate texture
+    const tex = model.framebuffer.getColorTexture();
+    tex.activate();
+    model.convolutionShader.setUniformi('u_image', tex.getTextureUnit());
+    model.convolutionShader.setUniform2f(
+      'u_textureSize',
+      tex.getWidth(),
+      tex.getHeight()
+    );
+    model.convolutionShader.setUniformfv('u_kernel', model.kernel);
+    model.convolutionShader.setUniformf(
+      'u_kernelWeight',
+      publicAPI.computeKernelWeight(model.kernel)
+    );
+
+    // render quad
+    gl.drawArrays(gl.TRIANGLES, 0, model.tris.getCABO().getElementCount());
+    tex.deactivate();
+  };
+
+  publicAPI.getFragmentShaderCode = (kernelDimension) => {
+    // generate new shader code
+    const kernelLength = kernelDimension * kernelDimension;
+    let shaderCode = [
+      '//VTK::System::Dec',
+      '//VTK::Output::Dec',
+      'uniform sampler2D u_image;',
+      'uniform vec2 u_textureSize;',
+      `uniform float u_kernel[${kernelLength}];`,
+      'uniform float u_kernelWeight;',
+      'varying vec2 tcoord;',
+      'void main(){',
+      '    vec2 onePixel = vec2(1.0, 1.0) / u_textureSize;',
+      '    vec4 colorSum =\n',
+    ].join('\n');
+
+    const halfDim = Math.floor(kernelDimension / 2);
+
+    // generate sum per pixel
+    let i = 0;
+    for (let y = -halfDim; y <= halfDim; ++y) {
+      for (let x = -halfDim; x <= halfDim; ++x) {
+        shaderCode += `        texture2D(u_image, tcoord + onePixel * vec2(${x}, ${y})) * u_kernel[${i}]`;
+        ++i;
+
+        if (i !== kernelLength) {
+          shaderCode += ' +\n';
+        }
+      }
+    }
+
+    // finish code
+    shaderCode += [
+      ';',
+      '    gl_FragData[0] = vec4((colorSum / u_kernelWeight).rgb, 1);',
```

Comment:
Why not keep the alpha value from `u_image` ?

Reply:
We will try it.